### DISCUSSION TITLE
WSL2/Docker experience isn't great on Windows...

### DISCUSSION THREAD

**Discussion by MartinHughes-BPC** [laravel/framework#35613] [General]
Hi all,

I don't know if this is the right place, but I have a bit of feedback on the Getting started docs for Windows, specifically my experience with installing and using WSL2/Docker/Laravel etc.

Docker-Desktop installed with no problems.
I had a few issues installing WSL2, related to having to upgrade Windows and having to do the additional steps that Windows Insiders don't.
Once I had everything up and running after multiple reboots, I followed the Laravel docs to get things installed. For whatever reason I kept getting messages about docker-composer being missing, which appeared to be a Python script. Eventually these messages disappeared of their own volition. I have no idea why.
That out of the way, my new Laravel 8 app worked... but incredibly slowly. I was getting response times on the default Welcome page of upwards of 1 second.
Went to bed.
Booted my machine today and got various messages from Docker about the backend exiting unexpectedly. Googled around for a bit and found that restarting the Windows WSL service (which is called something completely different and unobvious) - several times - sorted it.
I decided to take a break and boot up a VM... only to find VirtualBox running like a dog. And I mean a dead dog. Slow doesn't describe it. Googled again and yes, on Windows, there may be issues running WSL, it hogging Hyper-V, VirtualBox and a hundred - perhaps a thousand! - other things I don't understand.

You see that painting by Edvard Munch, _The Scream_? That.

To be clear I'm not accusing any party of anything, but simply reporting the facts as I found them. But if somebody new to Laravel asked me if the Getting started advice was any good, I'd tell them to try something else.

Cheers,
Martin.

**Comment by donnysim** [comment]
The performance of WSL2 across operating systems drives is a known issue. I would not recommend using WSL2 unless you move all your projects into the Linux drive and use PHPStorm IDE inside Linux via display server (but even then the experience is not great compared to running it directly) or use VSCode (which has support for WSL2 but I don't use VSCode so don't know much). We can only hope the situation will improve but for the time being it's more of a hassle than it's worth.
